examples/ui_actions_explorer/public/plugin.tsx

```diff
@@ -67,29 +76,26 @@ export class UiActionsExplorerPlugin implements Plugin<void, void, {}, StartDeps
     deps.uiActions.registerTrigger({
       id: USER_TRIGGER,
     });
-    deps.uiActions.registerAction(lookUpWeatherAction);
-    deps.uiActions.registerAction(viewInMapsAction);
-    deps.uiActions.registerAction(makePhoneCallAction);
-    deps.uiActions.registerAction(showcasePluggability);
 
     const startServices = core.getStartServices();
-    deps.uiActions.registerAction(
+
+    deps.uiActions.attachAction(
+      USER_TRIGGER,
       createPhoneUserAction(async () => (await startServices)[1].uiActions)
     );
-    deps.uiActions.registerAction(
+    deps.uiActions.attachAction(
+      USER_TRIGGER,
       createEditUserAction(async () => (await startServices)[0].overlays.openModal)
     );
-    deps.uiActions.attachAction(USER_TRIGGER, PHONE_USER_ACTION);
-    deps.uiActions.attachAction(USER_TRIGGER, EDIT_USER_ACTION);
 
     // What's missing here is type analysis to ensure the context emitted by the trigger
     // is the same context that the action requires.
-    deps.uiActions.attachAction(COUNTRY_TRIGGER, VIEW_IN_MAPS_ACTION);
-    deps.uiActions.attachAction(COUNTRY_TRIGGER, TRAVEL_GUIDE_ACTION);
-    deps.uiActions.attachAction(COUNTRY_TRIGGER, SHOWCASE_PLUGGABILITY_ACTION);
-    deps.uiActions.attachAction(PHONE_TRIGGER, CALL_PHONE_NUMBER_ACTION);
-    deps.uiActions.attachAction(PHONE_TRIGGER, SHOWCASE_PLUGGABILITY_ACTION);
-    deps.uiActions.attachAction(USER_TRIGGER, SHOWCASE_PLUGGABILITY_ACTION);
+    deps.uiActions.attachAction(COUNTRY_TRIGGER, viewInMapsAction);
```

Comment:
Here I tried to replace `viewInMaps` for `showcasePluggability` and typescript didn't complain.
I run `node scripts/type_check --project examples/ui_actions_explorer/tsconfig.json` to make sure it is not my IDE problem

Reply:
it is expected. `showcasePluggability` doesn't require any context so it can be attached to any trigger.  It's okay to attach an action to a trigger that emits _more_ than the action requires.

Reply:
Ok, this makes sense, thanks.

How about this example, I find it particularly weird: 

I change it to: 

```
deps.uiActions.attachAction(USER_TRIGGER, lookUpWeatherAction);
```

And see the ts error message. "can't assign string to Partial<UserContext> | undefined" which is great. 

But then I went and changed UserContext from `type Country = string;` to `type Country = {country: string}`


And this no longer throws typescript error for me 🤔 
```
deps.uiActions.attachAction(USER_TRIGGER, lookUpWeatherAction);
```

It seems that it should complain in this case. Not sure if this is me or the current types are not working well for matching between different object shapes because of `Partial`.

Reply:
Awesome catch!  It was an issue with `Partial`.  I'm honestly not sure why it works now, but it seems like requiring an object let me get rid of partial type and now it works.  Strange, but when I tested, (I changed the trigger context shape to `CountryContext & { extra: string }`) it seems to work now:

<img width="1445" alt="Screen Shot 2020-03-02 at 4 27 50 PM" src="https://user-images.githubusercontent.com/16563603/75719730-49743a80-5ca3-11ea-9b5e-806fef57f406.png">

Reply:
Awesome! seems like working as expected 👍